blog/2021-06-17-activity-timeouts.md

```diff
@@ -0,0 +1,242 @@
+
+If you let it, Temporal will retry a failing activity for up to 10 years! (with exponential backoff up to a defined maximum interval)
+Most Temporal developers will want to finetune how their system retries to balance user experience against the unreliability of the activity.
+
+While you can control intervals between retries and maximum number of retries in the `RetryPolicy`, `ScheduleToClose` is the best way to control retries based on *time elapsed*. We recommend using `ScheduleToClose` to limit retries rather than tweaking the number of `MaximumAttempts`, because that more closely matches desired user experience in the majority of cases.
+
+## Heartbeat Timeout
+
+For long running activities, we recommend recording heartbeats to create more frequent pingbacks from the Activity Worker to Temporal Server. 
+Then, set a `HeartbeatTimeout` to create a failure when Temporal fails to receive a heartbeat from your activity at the expected frequency.
+
+![image](https://user-images.githubusercontent.com/6764957/122424268-287f4800-cfc1-11eb-8417-f52743347afc.png)
+
+By their nature, Heartbeats must be recorded from Activity code using SDK APIs:
+
+<Tabs
+  defaultValue="go"
+  values={[
+    {label: 'Go', value: 'go'},
+    {label: 'Java', value: 'java'},
+  ]
+}>
+
+<TabItem value="go">
+
+```go
+progress := 0
+for hasWork {
+    // Send heartbeat message to the server.
+    activity.RecordHeartbeat(ctx, progress)
+    // Do some work.
+    ...
+    progress++
+}
+```
+
+</TabItem>
+<TabItem value="java">
+
+```java
+while ((read = inputStream.read(bytes)) != -1) {
+  totalRead += read;
+  f.write(bytes, 0, read);
+  // Let the Server know about the download progress.
+  Activity.getExecutionContext().heartbeat(totalRead);
+}
+```
+
+</TabItem>
+</Tabs>
+
+Setting a `HeartbeatTimeout` allows us to retry activities more quickly (e.g. the next minute after a heartbeat is missed) than the `StartToCloseTimeout`, which is necessarily set to as long as the longest possible activity (e.g. five hours later when we are sure the activity should have been completed).
+
+There are some minor nuances to heartbeats that may be of interest:
+
+- You can freely record heartbeats as often as you want - once a minute, or everytime a loop iterates - the SDKs throttle the heartbeats that get sent back anyway.
+- If a `HeartbeatTimeout` isn't set and the activity tries to record one, nothing will be recorded since that information will never be used.
+
+## ScheduleToStart Timeout
+
+We use the `ScheduleToStartTimeout`  to set a maximum limit that an activity should sit in a queue. We recommend that most users monitor the `ScheduleToStart` latency metric and set alerts for them as a [production scaling](https://docs.temporal.io/docs/server/production-deployment/#faq-autoscaling-workers-based-on-task-queue-load) metric, rather than setting an explicit timeout for it.
```

Comment:
```suggestion
The Schedule-To-Start Timeout  sets a limit on the amount of time that an Activity Task can sit in a Task Queue.
We recommend that most users monitor the `ScheduleToStart` latency metric and set alerts for that as a [production scaling](https://docs.temporal.io/docs/server/production-deployment/#faq-autoscaling-workers-based-on-task-queue-load) metric, rather than setting an explicit timeout for it.
```
> `ScheduleToStart` latency metric

How is that actually emitted? `schedule-to-start` all lower case I am guessing?